### ISSUE TITLE
How to install this package?

### ISSUE THREAD

**Issue by jurgenhaas** [scheeser/evernote-edit#14]
This sounds very promissing and I'm very keen to get this to work and to provide feedback. But I'm not sure how to install.

What I did was to clone the project into ~/.atom/packages and restarted Atom. The evernote package shows up in the package list and also in the menu under Packages -> Evernote Edit. When I select "Open Note", nothing happens. Then I looked into the console of Atom and found the error message "Failed to activate package named 'evernote-edit' Error: Cannot find module 'evernote'". Looks like my system is missing something and I wonder what it is and where to get it from.

Not, I'm on Ubuntu 14.04 and there is no usable Evernote client, this is why Evernote inside Atom is so promissing to me.

**Comment by scheeser** [comment]
Sure, it seems like the problem you are running into is that the node modules the plugin uses are not installed by simply cloning the repo. This typically happens during install from apm but since I haven't published my package that doesn't happen automatically.

So I would do something like:
- clone the directory
- open a terminal window and run `apm link` this should create a symlink from the package directory where you cloned the project to the atom packages file
  - `apm link -h` may be valuable to read
  - You may want to use the --dev option and open atom in dev mode to avoid messing anything up with your primary atom packages
- Run `apm install` from the plugin directory.
  - This step will download any of the node modules the plugin requires.

This was sort of edited from an outdated [support document](https://atom.io/docs/v0.84.0/contributing) on the atom site.

As an FYI to open a note right now you need either a evernote uid or the URL link for the note itself.

**Comment by jurgenhaas** [comment]
Great, `apm install` within the evernote-edit directory did the trick. Now I'm getting a small dialog asking for a node link or GUID. I tried several things but none of the values seems to be working. Let me give you an example:

https://www.evernote.com/shard/s1/sh/8d449fc0-bb72-46d2-b51f-fae04b0f9867/5ecb8c7c3099929f29f11a1d241821a5

When I use that link the error message reads "Uncaught TypeError: Cannot read property 'match' of null"

**Comment by scheeser** [comment]
If you open the developer tools and look at the console can you provide the full stacktrace on the error?

You also uncovered another problem in that I can't open that shared note even though it's public.

**Comment by jurgenhaas** [comment]
Here you go, this is the full stack trace:

```
Uncaught TypeError: Cannot read property 'match' of null /home/jurgenhaas/.atom/packages/evernote-edit/node_modules/evernote/evernote-sdk-js/evernote/node/c…:190
(anonymous function) /home/jurgenhaas/.atom/packages/evernote-edit/node_modules/evernote/evernote-sdk-js/evernote/node/c…:190
(anonymous function) /home/jurgenhaas/.atom/packages/evernote-edit/node_modules/evernote/evernote-sdk-js/evernote/node/c…:65
Store.getThriftClient /home/jurgenhaas/.atom/packages/evernote-edit/node_modules/evernote/evernote-sdk-js/evernote/node/c…:189
(anonymous function) /home/jurgenhaas/.atom/packages/evernote-edit/node_modules/evernote/evernote-sdk-js/evernote/node/c…:157
(anonymous function) /home/jurgenhaas/.atom/packages/evernote-edit/node_modules/evernote/evernote-sdk-js/evernote/node/c…:78
Store.getThriftClient /home/jurgenhaas/.atom/packages/evernote-edit/node_modules/evernote/evernote-sdk-js/evernote/node/c…:189
(anonymous function) /home/jurgenhaas/.atom/packages/evernote-edit/node_modules/evernote/evernote-sdk-js/evernote/node/c…:157
(anonymous function) /home/jurgenhaas/.atom/packages/evernote-edit/lib/open-note.js:30
_fulfilled /opt/atom/resources/app/node_modules/q/q.js:787
self.promiseDispatch.done /opt/atom/resources/app/node_modules/q/q.js:816
Promise.promise.promiseDispatch /opt/atom/resources/app/node_modules/q/q.js:749
(anonymous function) /opt/atom/resources/app/node_modules/q/q.js:557
flush /opt/atom/resources/app/node_modules/q/q.js:108
_tickCallback node.js:349
```

**Comment by scheeser** [comment]
I think you are running into a problem because you authorization token isn't set: https://dev.evernote.com/doc/articles/authentication.php#devtoken

There is an option to set this in settings or you can put a evernote-edit.txt file in your .atom folder in your user directory if say...you sync your settings up with a remote repository.

**Comment by jurgenhaas** [comment]
Right, that field wasn't available before I apm install'ed and afterwards I wasn't checking anymore. I've no included my token and get this error instead:

Uncaught TypeError: Cannot read property 'content' of undefined /home/jurgenhaas/.atom/packages/evernote-edit/lib/open-note.js:32

**Comment by jurgenhaas** [comment]
Ah, wait. That token is for the sandbox environment and of course the note URL isn't available there. I'll have to try and get a production API token first.

**Comment by scheeser** [comment]
There is a setting (checkbox) to use the sandbox environment in settings if you did want to test some things outside of your personal account.

**Comment by jurgenhaas** [comment]
Yeah, but my note is on their production server, so I got them to activate my API key and I included the API secret into the settings in atom. Still, I'm getting the error `Uncaught TypeError: Cannot read property 'content' of undefined /home/jurgenhaas/.atom/packages/evernote-edit/lib/open-note.js:32`without a stack trace.

What exactly do I have to input for Open Note? Is it the full URL or which part of the URL is meant to be used?

**Comment by scheeser** [comment]
The token to use isn't a oauth key/secret pair. It's a developer token which is different. See the link I posted above for details. In production you can visit https://www.evernote.com/api/DeveloperToken.action login copy and paste that value into the settings. The sublime text Evernote plugin has a decent [document](https://github.com/bordaigorl/sublime-evernote/wiki/First-Use) on the topic since it uses the same scheme for authorization.

As far as opening the note, URL should be sufficient or if you know the note guid it can be used as well.

Hopefully, some better error handling, readme and authorization workflow may help this confusion.

**Comment by jurgenhaas** [comment]
Great, this is indeed working. Both opening an saving such nodes is working just fine.

Are you planning to browser the note store eventually?

**Comment by scheeser** [comment]
Most likely. Can't give a timeline but having a browser of some kind would be an improvement.
